sw/host/opentitanlib/src/otp/alert_handler.rs

```diff
@@ -0,0 +1,511 @@
+            _ => AlertClass::X,
+        }
+    }
+}
+
+#[derive(TryFromPrimitive)]
+#[repr(u8)]
+enum AlertEnable {
+    None = 0xa9,
+    Enabled = 0x07,
+    Locked = 0xd2,
+}
+
+#[derive(TryFromPrimitive)]
+#[repr(u8)]
+enum AlertEscalate {
+    None = 0xd1,
+    Phase0 = 0xb9,
+    Phase1 = 0xcb,
+    Phase2 = 0x25,
+    Phase3 = 0x76,
+}
+
+struct AlertClassConfig {
+    enabled: AlertEnable,
+    escalate: AlertEscalate,
+    accum_thresh: u32,
+    timeout_cyc: u32,
+    phase_cycs: [u32; ALERT_HANDLER_PARAM_N_PHASES as usize],
+}
+
+impl Default for AlertClassRegs {
+    fn default() -> Self {
+        AlertClassRegs {
+            regwen: 1,
+            ctrl: 0,
+            accum_thresh: 0,
+            timeout_cyc: 0,
+            phase_cycs: [0; ALERT_HANDLER_PARAM_N_PHASES as usize],
+        }
+    }
+}
+
+impl Default for AlertRegs {
+    fn default() -> Self {
+        AlertRegs {
+            regwen: [1; ALERT_HANDLER_ALERT_REGWEN_MULTIREG_COUNT as usize],
+            loc_regwen: [1; ALERT_HANDLER_LOC_ALERT_REGWEN_MULTIREG_COUNT as usize],
+            en: [0; ALERT_HANDLER_ALERT_EN_SHADOWED_MULTIREG_COUNT as usize],
+            class: [0; ALERT_HANDLER_ALERT_CLASS_SHADOWED_MULTIREG_COUNT as usize],
+            loc_en: [0; ALERT_HANDLER_LOC_ALERT_EN_SHADOWED_MULTIREG_COUNT as usize],
+            loc_class: [0; ALERT_HANDLER_LOC_ALERT_CLASS_SHADOWED_MULTIREG_COUNT as usize],
+            class_regs: [Default::default(); ALERT_HANDLER_PARAM_N_CLASSES as usize],
+        }
+    }
+}
+
+impl AlertRegs {
+    /// Compute the CRC32 of the internal register values to match the value produced by
+    /// `sw/device/silicon_creator/lib/drivers/alert.h:alert_config_crc32`.
+    pub fn crc32(self) -> u32 {
+        let crc = new_crc();
+        let mut digest = crc.digest();
+        self.crc32_add(&mut digest);
+        digest.finalize()
+    }
+
+    /// Create the set of alert_handler register values from a given lifecycle state and OTP.
+    ///
+    /// The internal fields of `AlertRegs` should match those produced on the device after
+    /// alert_handler is configured in `sw/device/silicon_creator/lib/shutdown.h:shutdown_init`.
+    pub fn try_new<T: OtpRead>(lc_state: LcStateVal, otp: &T) -> Result<Self> {
+        let mut alert = AlertRegs::default();
+
+        let lc_shift = match lc_state {
+            LcStateVal::Prod => 0,
+            LcStateVal::ProdEnd => 8,
+            LcStateVal::Dev => 16,
+            LcStateVal::Rma => 24,
+            LcStateVal::Test => return Ok(alert),
+        };
+
+        let class_enable = otp.read32("ROM_ALERT_CLASS_EN")?;
+        let class_escalate = otp.read32("ROM_ALERT_ESCALATION")?;
+
+        for i in 0..ALERT_HANDLER_ALERT_CLASS_SHADOWED_MULTIREG_COUNT as usize {
+            let value = otp.read32_offset(Some("ROM_ALERT_CLASSIFICATION"), i * 4)?;
+            let cls = AlertClass::try_from(
+                value.view_bits::<Lsb0>()[lc_shift..lc_shift + 8].load_le::<u8>(),
```

Comment:
nit: Sorry for nitpicking but since we are accessing things in byte granularity, can we use sthg like `value.to_le_bytes()[byte_index]`?